From 45809561c5a480d2b1e506d9baa67ec67168118a Mon Sep 17 00:00:00 2001
From: Gurdal Oruklu <gurdaloruklu@gmail.com>
License: GPLV2
Source: COMCAST
Upstream-Status: none
Date: Tue, 10 Sep 2019 18:07:02 +0000
Subject: [PATCH] updated platform player supportsType() function to support
 additional mimetype arguments (cryptoblockformat & eotf) Youtube requires
 cryptoblockformat support is added in this commit in upstream WPE:
 https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/0e4b75d3a2f3ae7a2fad52879fc6955fd6d9081b

Signed-off-by: Gurdal Oruklu <gurdaloruklu@gmail.com>
---
 .../mse/MediaPlayerPrivateGStreamerMSE.cpp         | 32 ++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
index 2391535..20796d2 100644
--- a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
@@ -1148,6 +1148,8 @@ MediaPlayer::SupportsType MediaPlayerPrivateGStreamerMSE::supportsType(const Med
         return result;
     }
 
+    GST_DEBUG("supportsType() %s", parameters.type.raw().utf8().data());
+
     bool ok;
     unsigned channels = parameters.type.parameter(ASCIILiteral("channels")).toUInt(&ok);
     if (ok && channels > MEDIA_MAX_AAC_CHANNELS)
@@ -1168,6 +1170,34 @@ MediaPlayer::SupportsType MediaPlayerPrivateGStreamerMSE::supportsType(const Med
     if (ok && framerate > MEDIA_MAX_FRAMERATE)
         return result;
 
+    // YT check for  electro-optic transfer function (EOTF) support
+    // Possible values:
+    // bt709 (SDR)
+    // smpte2084 HDR10
+    // arib-std-b67 HLG
+
+    String eotf = parameters.type.parameter(ASCIILiteral("eotf"));
+    if (!eotf.isEmpty())
+    {
+#if ENABLE(VP9_HDR)
+        if(eotf == "bt709"){
+            GST_DEBUG("eotf = bt709");
+        }else if(eotf == "smpte2084"){
+            GST_DEBUG("eotf = smpte2084");
+        }else if(eotf == "arib-std-b67"){
+            GST_DEBUG("eotf = arib-std-b67");
+        }else{
+            GST_WARNING("unsupported eotf: %s", eotf.utf8().data());
+            return result;
+        }
+#else
+        if(eotf != "bt709"){
+            GST_WARNING("unsupported eotf: %s", eotf.utf8().data());
+            return result;
+        }
+#endif
+    }
+
     // Spec says we should not return "probably" if the codecs string is empty.
     if (MediaPlayerPrivateGStreamerMSE::mimeTypeCache().contains(containerType)) {
         Vector<String> codecs = parameters.type.codecs();
@@ -1177,6 +1207,8 @@ MediaPlayer::SupportsType MediaPlayerPrivateGStreamerMSE::supportsType(const Med
             result = supportsAllCodecs(codecs) ? MediaPlayer::IsSupported : MediaPlayer::IsNotSupported;
     }
 
+    GST_DEBUG("supportsType() result: %d", result);
+
     return extendedSupportsType(parameters, result);
 }
 
-- 
1.9.1

